ts/src/transports/Transport.ts

```diff
@@ -19,6 +21,31 @@ export type TransportOptions = {
   onComplete: (err?: Error) => void,
 }
 
+let xhr: XMLHttpRequest;
+function getXHR () {
+  if (xhr !== undefined) return xhr;
+
+  if (XMLHttpRequest) {
+    xhr = new XMLHttpRequest();
+    try {
+      xhr.open('GET', 'https://example.com')
```

Comment:
This can be `https://localhost`. The transport feature detection requires an `XMLHttpRequest`, which can only be constructed with a valid URL. It doesn't actually make a request.